includes/class-sensei-frontend.php

```diff
@@ -139,9 +139,11 @@ public function enqueue_scripts() {
 
 			$suffix = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? '' : '.min';
 
-			// My Courses tabs script.
-			wp_register_script( Sensei()->token . '-user-dashboard', esc_url( Sensei()->plugin_url . 'assets/js/user-dashboard' . $suffix . '.js' ), array( 'jquery-ui-tabs' ), Sensei()->version, true );
-			wp_enqueue_script( Sensei()->token . '-user-dashboard' );
+			if ( is_page( Sensei()->settings ? (int) Sensei()->settings->settings['my_course_page'] : -1 ) ) {
```

Comment:
`Sensei()->settings` seems relatively meaningless, it's all but guaranteed to exist.

I would go with:

```php
if ( isset( Sensei()->settings->settings['my_course_page'] ) {
  if ( is_page( Sensei()->settings->settings['my_course_page'] ) {
    wp_enqueue_script( Sensei()->token . '-user-dashboard', esc_url( Sensei()->plugin_url . 'assets/js/user-dashboard' . $suffix . '.js' ), array( 'jquery-ui-tabs' ), Sensei()->version, true );
  }
}
```

It doesn't seem to be useful to separately register this script handle as it looks like a single-use script?

But if rest of the scripts all follow register/enqueue dance convention, it makes sense to follow that model.

Reply:
Yeah, I kind of assumed that `Sensei()->settings` should be present all the time, except the renderer tests were failing on Travis on PHP 5.6 (passing locally on PHP 7.3) -- I think because the `settings` don't get set in those tests? So I added that check in instead of changing the tests ... I'll see if your suggestion works better.